cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/DefaultSchedulerService.java

```diff
@@ -68,7 +57,7 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
       String key = trigger.getKey().getName();
       String[] parts = key.split(":");
       Preconditions.checkArgument(parts.length == 6, String.format("Trigger's key name %s has %d parts instead of 6",
```

Comment:
The key is actually not ScheduleId, it's a ProgramId ending with a schedule name. Also the programType here is a String generated from `SchedulableProgramType` instead from  `ProgramType`.